src/main/java/uk/gov/nationalarchives/pdi/step/jena/shacl/JenaShaclStep.java

```diff
@@ -0,0 +1,85 @@
+package uk.gov.nationalarchives.pdi.step.jena.shacl;
+
+import org.apache.jena.graph.Graph;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.riot.RDFDataMgr;
+import org.apache.jena.shacl.ShaclValidator;
+import org.apache.jena.shacl.ValidationReport;
+import org.apache.jena.shacl.validation.ReportEntry;
+import org.pentaho.di.core.exception.KettleException;
+import org.pentaho.di.core.row.RowMetaInterface;
+import org.pentaho.di.trans.Trans;
+import org.pentaho.di.trans.TransMeta;
+import org.pentaho.di.trans.step.*;
+
+/**
+ *  A PDI plugin step which can be used to validate an Apache Jena Model contained in a row field against a SHACL shape
+ *  file loaded from a give path on the file system
+ */
+public class JenaShaclStep extends BaseStep implements StepInterface {
+
+    public JenaShaclStep(final StepMeta stepMeta, final StepDataInterface stepDataInterface, final int copyNr,
+                         final TransMeta transMeta, final Trans trans) {
+        super(stepMeta, stepDataInterface, copyNr, transMeta, trans);
+    }
+
+    @Override
+    public boolean processRow(final StepMetaInterface smi, final StepDataInterface sdi) throws KettleException {
+        final JenaShaclStepMeta meta = (JenaShaclStepMeta) smi;
+        final JenaShaclStepData data = (JenaShaclStepData) sdi;
+
+        final Object[] row = getRow();
+        if (row == null) {
+            setOutputDone();
+            return false;
+        }
+
+        if(first) {
+            first = false;
+            data.setValidator(ShaclValidator.get());
+            data.setShapesGraph(RDFDataMgr.loadGraph(meta.getShapesFilePath()));
```

Comment:
Can `RDFDataMgr.loadGraph` return null or throw an exception? We should maybe check/handle that?

Reply:
Updated and added test for this.